### PR TITLE
Add support for handling executor plugins

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add support for handling executor plugins
- Add gRPC resource generation
- Add e2e test for executor plugins. I tested that on my fork already: https://github.com/mszostok/botkube/actions/runs/3523896348
  -  more information about e2e testing https://github.com/kubeshop/botkube/blob/7bdbc042b28f04fbaa29fe2d51d084346c130fbf/docs/proposal/2022-11-22-botkube-plugins-integration.md#e2e-testing

Other:
- update goimport-reviser
- remove outdated `e2eTest` object from `e2e-test-values.yaml`

## Testing
I changed the e2e tests workflow, so it won't work on this PR. However, I tested that on my fork: https://github.com/mszostok/botkube/actions/runs/3523896348/jobs/5908734163

Install Botkube:
```
export SLACK_BOT_TOKEN="xoxb-"

helm upgrade botkube --install --namespace botkube ./helm/botkube --wait --create-namespace \
  -f ./helm/botkube/e2e-test-values.yaml \
  --set image.repository=kubeshop/pr/botkube \
  --set image.tag=862-PR \
  --set image.pullPolicy="Always" \
  --set communications.default-group.slack.token="${SLACK_BOT_TOKEN}"
```

Build plugins:
```
make build-plugins
```

Run tests:
```
export DISCORD_TESTER_APP_TOKEN="dummy"
export DISCORD_GUILD_ID="dummy"
export DISCORD_TESTER_NAME="dummy"

export KUBECONFIG="<config>"
export SLACK_BOT_NAME="botkube" # cannot be one for socket
export SLACK_TESTER_NAME="tester"
export SLACK_ADDITIONAL_CONTEXT_MESSAGE="local"
export SLACK_TESTER_APP_TOKEN="xoxb-"
export PLUGINS_BINARIES_DIRECTORY="dist"
make test-integration-slack
```

## Related issue(s)

- #838

### CODE REVIEW

**Comment by huseyinbabal** [comment]
Is it ok if I wait until proposal doc merged? I can review this PR with clearer understanding if I wait for final proposal version.

**Comment by mszostok** [comment]
Hi @huseyinbabal!

I thought that only minor things will be adjusted, so that's the reason why I put it into review. But let's finish the proposal as you mentioned to be sure that there won't be some bigger changes 👍

File: .goreleaser.plugin.yaml
Comment (huseyinbabal) [inline .goreleaser.plugin.yaml:17]:
```suggestion
      - arm64
    goarm:
      - 7
```

File: hack/goreleaser.sh
Comment (huseyinbabal) [inline hack/goreleaser.sh:96]:
```suggestion
```

Since prepare is needed for only docker build operation

File: internal/plugin/collector.go
Comment (huseyinbabal) [inline internal/plugin/collector.go:53]: Would it be insightful if we log the disabled plugins? At least in debug mode.

File: internal/plugin/collector.go
Comment (huseyinbabal) [inline internal/plugin/collector.go:56]: Would it be insightful if we log the not binded plugins? At least in debug mode.

File: internal/plugin/index.go
Comment (huseyinbabal) [inline internal/plugin/index.go:22]: Remember we realized the cache is not refreshed even we added new plugins. We would implement 2 options;
- Use etag and if-modified-since if github provides to refresh our cache as a first step once we fetched the index during startup
- Since you already a Index data structure, you can implement a function Exists that checks plugin, if not exist, you can refresh cache
- Or you can simply retry index fetch if you get non-ok from plugin fetch response

**Comment by huseyinbabal** [comment]
@mszostok finished review, thank you for the valuable work 🚀 I added minor suggestion, you can ping me if you need re-review or approve.

File: internal/plugin/index.go
Comment (huseyinbabal) [reply internal/plugin/index.go:22]: the simpler the better for me, thanks :)

**Review by huseyinbabal — APPROVED** [review]

**Pull request merged**
